### PR TITLE
[TEST] Add coverage to sequence file format fasta test

### PR DESCRIPTION
**Pull request by MitraDarja**
Part of https://github.com/seqan/product_backlog/issues/222

### CODE REVIEW

File: test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp [hunk 1]
```diff
@@ -180,6 +180,33 @@ TEST_F(read, mixed_issues)
...
 }
 
+TEST_F(read, fail_no_newline_after_id)
+{
+    std::string input;
+    std::stringstream istream{input =
+    {
+        ">ID1"
+        "ACGTTTTTTTTTTTTTTT"
+    }};
```
Comment (eseiler) [inline test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp:190]:
```suggestion
    std::string input
    {
        ">ID1"
        "ACGTTTTTTTTTTTTTTT"
    };
    std::stringstream istream{input};
```

File: test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp [hunk 2]
```diff
@@ -180,6 +180,33 @@ TEST_F(read, mixed_issues)
...
+}
+
+TEST_F(read, fail_no_newline_after_truncate_id)
+{
+    std::string input;
+    std::stringstream istream{input =
+    {
+        ">ID1"
+        "ACGTTTTTTTTTTTTTTT"
+    }};
```
Comment (eseiler) [inline test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp:203]:
```suggestion
    std::string input
    {
        ">ID1"
        "ACGTTTTTTTTTTTTTTT"
    };
    std::stringstream istream{input};
```

**Review by eseiler — CHANGES_REQUESTED** [review]

**Review by eseiler — APPROVED** [review]

File: test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp [hunk 3]
```diff
@@ -180,6 +180,31 @@ TEST_F(read, mixed_issues)
...
+    std::stringstream istream{input};
+
+    seqan3::sequence_file_input fin{istream, seqan3::format_fasta{}};
+    EXPECT_THROW(fin.begin(), seqan3::unexpected_end_of_input);
+}
+
+TEST_F(read, fail_no_newline_after_truncate_id)
+{
+    std::string input{
+        ">ID1"
```
Comment (smehringer) [inline test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp:198]:
```suggestion
        ">ID1 to_be_truncated"
```

**Review by smehringer — CHANGES_REQUESTED** [review]
Just one small thing because truncating ids would best be tested with something to truncate I guess

**Comment by smehringer** [comment]
well that's a bug then. Good thing we added a test. I sent you a patch on gitter that you can apply (I would do a separate [FIX] commit prior to the one adding the test)

**Review by smehringer — CHANGES_REQUESTED** [review]
Looks good! 
Can you just clean the commit history? It should look something like this:
`[FIX] Check if sequence is present when reading.`
`[TEST] Add fasta format test for no-newline--after-id-error.`

The Fix has to be *before* the Test such that the history is always "buildable" :)
``

**Review by smehringer — APPROVED** [review]
Perfect thanks :)

**Pull request merged**